[SYCL] End-to-end tests for queue shortcut functions

andylshort opened the pull request:
End-to-end tests for each of the 7 explicit copy queue shortcut functions defined in Table 29 of [Section 4.6.5.2 of the specification](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:queue-shortcuts). 

This PR is contingent upon [intel/llvm#7266](https://github.com/intel/llvm/pull/7266) being accepted and merged.

Commit:
End-to-end tests for queue shortcut functions

andylshort commented:
/verify with https://github.com/intel/llvm/pull/7266

Commit:
clang-format'ed the tests

romanovvlad commented on SYCL/Basic/queue/queue_shortcut_functions.cpp:
```diff
@@ -0,0 +1,167 @@
+    Q.wait();
+
+    for (int i = 0; i < Size; ++i) {
+      if (Data[i] != FillValue) {
```
It should be illegal to access `Data` while `DataBuffer` is alive.

Commit:
Fixed data access issue with scoping

It's illegal to access data stored while the controlling sycl::buffer is
still alive, so I moved the sycl-based accesses to their own scope.

Commit:
Formatted with clang-format

andylshort commented:
https://github.com/intel/llvm/pull/7266 has now been merged into the sycl branch so this should now build in CI.

AlexeySachkov commented:
> [intel/llvm#7266](https://github.com/intel/llvm/pull/7266) has now been merged into the sycl branch so this should now build in CI.

You need to restart CI for that. You can either close and reopen the PR or simply merge `intel` branch into your PR to introduce an update here

AlexeySachkov commented on SYCL/Basic/queue/queue_shortcut_functions.cpp:
```diff
@@ -0,0 +1,165 @@
+  }
+
+  {
+    std::vector<int> ReferenceData = {1, 2, 3, 4, 5};
```
```suggestion
    const std::vector<int> ReferenceData = {1, 2, 3, 4, 5};
```

AlexeySachkov commented on SYCL/Basic/queue/queue_shortcut_functions.cpp:
```diff
@@ -0,0 +1,165 @@
+  }
+
+  {
+    std::vector<int> ReferenceData = {2, 4, 6, 8, 10};
```
```suggestion
    const std::vector<int> ReferenceData = {2, 4, 6, 8, 10};
```

AlexeySachkov commented on SYCL/Basic/queue/queue_shortcut_functions.cpp:
```diff
@@ -0,0 +1,165 @@
+  }
+
+  {
+    std::vector<int> ReferenceData = {3, 6, 9, 12, 15};
```
```suggestion
    const std::vector<int> ReferenceData = {3, 6, 9, 12, 15};
```

This might require turning `buffer<int>` into `buffer<const int>`

AlexeySachkov reviewed (approved):
Overall LGTM

Commit:
Merge branch 'intel' into alamzeds/queue-shortcuts-sycl2020

Commit:
Made reference data const so can't be changed
